### ISSUE TITLE
bug: Can't identify the selected text due to document highlights

### ISSUE THREAD

**Issue by s1n7ax** [rose-pine/neovim#277]
### Neovim version (nvim -v)

v0.11.0-dev+0e3e1e6

### Terminal / multiplexer

Alacritty

### Describe the bug

I have created a bug in LazyVim but it seems it's an issue in the colorscheme. You can find more information here.

https://github.com/LazyVim/LazyVim/issues/3785

**Comment by mvllow** [comment]
I'm not familiar with LazyVim but you should be able to override via:

```lua
require("rose-pine").setup({
  highlight_groups = {
    GroupName = { fg = "text", bg = "pine" } -- Or use a hex value
  }
})
```

Happy to accept a PR that adds support for this in the theme :)

**Comment by s1n7ax** [comment]
@mvllow I found the issue.

`LspReferenceText` and `Visual` both using `palette.highlight_med`. If we can make a difference, then it's easy to identify because both highlights could exist at the same time.

**Comment by mvllow** [comment]
Appreciate the info. I see two solutions here:

1. Change Visual to "overlay". I tried this locally and it looks fine in my opinion. It matches CursorLine but I'm not sure if that's important since the cursorline doesn't seem to appear in visual mode. Not able to test how "highlight_med" looks with this, though.
2. For higher contrast, we could use an accent color as the background, e.g. `LspReferenceText = { bg = "gold", fg = "text" }`.

If you're able to try either of these (or any other thoughts you may have) and share some screenshots of your favourite, I'm happy to implement it.

**Comment by s1n7ax** [comment]
@mvllow I like the first with `Visual = { bg = palette.overlay }`.

![image](https://github.com/rose-pine/neovim/assets/18459807/fc65f848-354a-4356-84f8-d248cb8d8025)

May be just a little contrasty `fg` gold color. 
Left: `Visual = { bg = palette.overlay }` 
Right: `Visual = { fg = "#fab048", bg = palette.overlay }`
![image](https://github.com/rose-pine/neovim/assets/18459807/06d02e2a-23e8-4c48-a7fd-e7db8d7ccd76)

**Comment by s1n7ax** [comment]
@mvllow I like the first with `Visual = { bg = palette.overlay }`.

![image](https://github.com/rose-pine/neovim/assets/18459807/fc65f848-354a-4356-84f8-d248cb8d8025)

May be just a little contrasty `fg` gold color. 
Left: `Visual = { bg = palette.overlay }` 
Right: `Visual = { fg = "#fab048", bg = palette.overlay }`

![image](https://github.com/rose-pine/neovim/assets/18459807/06d02e2a-23e8-4c48-a7fd-e7db8d7ccd76)

**Comment by mvllow** [comment]
Thank you for the screenshots :) I do like the Visual being set to "overlay". The contrast isn't ideal but can be improved in the future—having something distinctive, even a little, is worth it.

I've played around with using different opacities of our "muted" colour to replace all of our highlight colours which may give us more flexibility for cases where we need more shades for a neutral background. I think you could test these via `Visual = { bg = "muted", blend = 10 }` etc.

If I have time, I'll try the shades of "muted". Otherwise we'll go with "overlay" for Visual :)

**Comment by s1n7ax** [comment]
@mvllow `Visual = { bg = "muted", blend = 10 }` is very easy on my eye compared to `overlay`. In both cases for some reason `highlight_med` looks elevated like there is a drop shadow in my brain and Visual is not which is bit weird.

![image](https://github.com/rose-pine/neovim/assets/18459807/5a8e3f36-29d5-4c21-9ea9-7411142b2f2c)

Inverse of highlight seems appealing in visual mode
![image](https://github.com/rose-pine/neovim/assets/18459807/0a98d1c4-67a9-45be-bc5a-07f09dccb375)

But looks weird when in normal mode with cursor line

![image](https://github.com/rose-pine/neovim/assets/18459807/7489d0cd-5c75-4bd7-bb8e-1fcc02bf38fa)

**Comment by mirsella** [comment]
hey
i find `Visual = { bg = "highlight_high" }` pretty good too
![image](https://github.com/rose-pine/neovim/assets/45905567/e0454a0a-9df1-4729-a741-21775d2fe963)

i was wondering how did you manage to use directly line like this `Visual = { bg = palette.overlay }` with lazyvim ?
because the rose-pin plugin is not yet loaded in the opts field. Using the config field ?

**Comment by mvllow** [comment]
@mirsella I would do `Visual = { bg = 'overlay' }` similar to your screenshot :) We do export our palette for use but honestly I've not had good luck using it directly. Possibly something we could improve in the future

**Comment by sjclayton** [comment]
@mvllow 

Any more thoughts on this? or are you still wanting to use `overlay`...

**Comment by mirsella** [comment]
@sjclayton no I'm good, i wasn't experienced with color scheme config in nvim, the method of just putting the name of the palette as string is perfectly fine !

**Comment by sjclayton** [comment]
@mirsella 

I was asking the maintainer because I wanted to know if the plan was to have this added as default behavior or not.

Then it would be applied so it'd fix the issue for everyone, and you wouldn't have to do it in your config.

**Comment by mvllow** [comment]
I think setting "Visual" to "overlay" is good :) You never know what else is linking to that group and if it will break things but life is short

**Comment by mvllow** [comment]
Oh, the cursorline is set to "overlay"—that's probably why we used something different. I'm trying out this for visual:

```lua
require("rose-pine").setup({
	highlight_groups = {
		Visual = { bg = "iris", blend = 15 },
	},
})

```

**Comment by qxuken** [comment]
"leafy search"  recipe works well:

```lua
LspReferenceText = { fg = 'text', bg = 'leaf', blend = 20, inherit = false },
LspReferenceRead = { fg = 'base', bg = 'leaf', inherit = false },
LspReferenceWrite = { fg = 'text', bg = 'leaf', blend = 20, inherit = false },
```

**Comment by mvllow** [comment]
I changed `Visual` to a blended iris. Does this fix your issue too @qxuken? Looks like your overrides are good but I'm curious if the default `LspReference*` groups still need updated or not.

**Comment by qxuken** [comment]
Yeah, it does fix my issue. I do see a difference when I'm hovering versus when I'm in visual mode. However, I will probably still use overrides, since the color changes are more dramatic.
